### PR TITLE
Refactor/sor service

### PR DESCRIPTION
**Pull request by franzns**
moving, renaming and removing unused stuff.
Also adding on chain query for v2 swaps

### CODE REVIEW

File: modules/sor/sor.gql [hunk 1]
```diff
@@ -7,13 +7,41 @@ extend type Query {
...
+    swaps: [GqlSwap!]!
+    swapAmount: String!
+    swapAmountForSwaps: String!
+    returnAmount: String!
+    returnAmountFromSwaps: String!
+    returnAmountConsideringFees: String!
+    tokenIn: String!
+    tokenOut: String!
+    marketSp: String!
+}
```
Comment (gmbronco) [inline modules/sor/sor.gql]: perhaps we could already remove fields they don't use:

marketSp
returnAmountConsideringFees
swapAmountForSwaps

they didn't confirm which fields they use, didn't they?

File: modules/sor/sor.service.ts [hunk 2]
```diff
@@ -1,54 +1,81 @@
...
-        const swap = await this.getSwap({ ...input, swapOptions: {} });
-        const emptyResponse = EMPTY_COWSWAP_RESPONSE(input.tokenIn, input.tokenOut, input.swapAmount);
+    async getCowSwaps(args: QuerySorGetCowSwapsArgs): Promise<GqlCowSwapApiResponse> {
+        console.log('getCowSwaps args', JSON.stringify(args));
+        const amountToken = args.swapType === 'EXACT_IN' ? args.tokenIn : args.tokenOut;
+        // Use TokenAmount to help follow scaling requirements in later logic
+        // args.swapAmount is RawScale, e.g. 1USDC should be passed as 1000000
+        const amount = await getTokenAmountRaw(amountToken, args.swapAmount, args.chain!);
+
+        const swap = await sorV2Service.getSwapResult({
```
Comment (gmbronco) [inline modules/sor/sor.service.ts:25]: V2 only?

File: modules/sor/sor.service.ts [hunk 3]
```diff
@@ -1,54 +1,81 @@
...
+    async getSorSwaps(args: QuerySorGetSwapsArgs): Promise<GqlSorGetSwapsResponse> {
+        console.log('getSorSwaps args', JSON.stringify(args));
+        const tokenIn = args.tokenIn.toLowerCase();
+        const tokenOut = args.tokenOut.toLowerCase();
+        const amountToken = args.swapType === 'EXACT_IN' ? tokenIn : tokenOut;
+        // Use TokenAmount to help follow scaling requirements in later logic
+        // args.swapAmount is HumanScale
+        const amount = await getTokenAmountHuman(amountToken, args.swapAmount, args.chain!);
+
+        const swap = await this.getComparingSwap({
```
Comment (gmbronco) [inline modules/sor/sor.service.ts:55]: i like the name vs getSwap

**Review by gmbronco — APPROVED** [review]

File: modules/sor/sor.gql (hunk 1 above)
Comment (franzns) [reply modules/sor/sor.gql]: agree, will remove once I know what they use

File: modules/sor/sor.service.ts (hunk 2 above)
Comment (franzns) [reply modules/sor/sor.service.ts:25]: yes, balancer chains dont support a V1

**Pull request merged**